service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -188,12 +188,7 @@ protected final void extractOptionalJsonPayload(final RoutingContext ctx) {
      */
     protected final void extractRequiredJsonArrayPayload(final RoutingContext ctx) {
         extractRequiredJson(ctx, body -> {
-            final var payload = body.getBodyAsJsonArray();
-            if (payload != null && payload.getList() == null) {
-                // work around eclipse-vertx/vert.x#2993
-                return null;
-            }
-            return payload;
+            return body.getBodyAsJsonArray();
```

Comment:
Are we sure we don't need this anymore?

Reply:
the bug is fixed : https://github.com/eclipse-vertx/vert.x/issues/2993
and hono updated to a more recent version of vertx (3.8.4)

Reply:
OK. Great.